api/metrics/prometheus_handler.go

```diff
@@ -0,0 +1,148 @@
+	results, warnings, err := pc.client.GetMetrics(ctx, appName, envName, componentName, duration, since)
+	if err != nil {
+		return nil, err
+	}
+	resources := getUsedResourcesByMetrics(ctx, results, durationValue, sinceValue)
+	resources.Warnings = warnings
+	log.Ctx(ctx).Debug().Msgf("Got used resources for application %s", appName)
+	return resources, nil
+}
+
+func getUsedResourcesByMetrics(ctx context.Context, results map[internal.QueryName]prometheusModel.Value, queryDuration time.Duration, querySince time.Duration) *applicationModels.UsedResources {
+	usedCpuResource := applicationModels.UsedResource{}
+	usedCpuResource.Min = getCpuMetricValue(ctx, results, internal.CpuMin)
+	usedCpuResource.Max = getCpuMetricValue(ctx, results, internal.CpuMax)
+	usedCpuResource.Avg = getCpuMetricValue(ctx, results, internal.CpuAvg)
+	usedMemoryResource := applicationModels.UsedResource{}
+	usedMemoryResource.Min = getMemoryMetricValue(ctx, results, internal.MemoryMin)
+	usedMemoryResource.Max = getMemoryMetricValue(ctx, results, internal.MemoryMax)
+	usedMemoryResource.Avg = getMemoryMetricValue(ctx, results, internal.MemoryAvg)
+	now := time.Now()
+	return &applicationModels.UsedResources{
+		From:   radixutils.FormatTimestamp(now.Add(-queryDuration)),
+		To:     radixutils.FormatTimestamp(now.Add(-querySince)),
+		CPU:    &usedCpuResource,
+		Memory: &usedMemoryResource,
+	}
+}
+
+func parseQueryDuration(duration string, defaultValue string) (time.Duration, string, error) {
+	if len(duration) == 0 || !regexp.MustCompile(durationExpression).MatchString(duration) {
+		duration = defaultValue
+	}
+	if len(duration) == 0 {
+		return 0, duration, nil
+	}
+	parsedDuration, err := prometheusModel.ParseDuration(duration)
+	return time.Duration(parsedDuration), duration, err
+}
+
+func getCpuMetricValue(ctx context.Context, queryResults map[internal.QueryName]prometheusModel.Value, queryName internal.QueryName) *float64 {
+	if value, ok := getMetricsValue(ctx, queryResults, queryName); ok {
+		return pointers.Ptr(value)
+	}
+	return nil
+}
+
+func getMemoryMetricValue(ctx context.Context, queryResults map[internal.QueryName]prometheusModel.Value, queryName internal.QueryName) *float64 {
+	if value, ok := getMetricsValue(ctx, queryResults, queryName); ok {
+		return pointers.Ptr(value)
+	}
+	return nil
+}
+
+func getMetricsValue(ctx context.Context, queryResults map[internal.QueryName]prometheusModel.Value, queryName internal.QueryName) (float64, bool) {
+	queryResult, ok := queryResults[queryName]
+	if !ok {
+		return 0.0, false
+	}
+	groupedMetrics, ok := queryResult.(model.Vector)
+	if !ok {
+		log.Ctx(ctx).Error().Msgf("Failed to convert metrics query %s result to Vector", queryName)
+		return 0, false
+	}
+	sum := 0.0
+	for _, sample := range groupedMetrics {
+		sum += float64(sample.Value)
+	}
+	return sum, true
+}
+
+func getPrometheusQueries(appName, envName, componentName, duration, since string) map[internal.QueryName]string {
```

Comment:
PErhaps we should move this function to prometheus_client.go since that is where it is used

Reply:
Moved